File: app/src/main/java/com/agotakiss/androidtest/presentation/BaseActivity.java
package com.agotakiss.androidtest.presentation;

import android.app.Activity;
import android.util.Log;

public class BaseActivity extends Activity {
    protected final String TAG = this.getClass().getSimpleName();


    protected void logE(Object message){
        Log.e(TAG, message.toString());
    }
    protected void logD(String message){
        Log.d(TAG, message);
    }
}

File: app/src/main/java/com/agotakiss/androidtest/presentation/BaseActivity.kt
package com.agotakiss.androidtest.presentation

import android.app.Activity
import android.content.ContentValues.TAG
import android.support.v7.app.AppCompatActivity
import android.util.Log
import com.agotakiss.androidtest.base.MovieApplication

abstract class BaseActivity : AppCompatActivity() {

    val Activity.movieApplication: MovieApplication get() = application as MovieApplication

    protected fun logE(message: Any) {
        Log.e(TAG, message.toString())
    }

    protected fun logD(message: String) {
        Log.d(TAG, message)
    }
}
